0000-always-run-model-hook.md

```diff
@@ -0,0 +1,81 @@
+- Start Date: 2017-12-07
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The model hook will always run, even for dynamic segments that already have their models loaded.
+
+# Motivation
+
+Currently the router will not run the model hook when transition to a route with a dynamic segment that already has its model loaded. This behavior is well intended, if we are using `{{link-to "post.show" post}}` then there's no reason to run the `post.show` model hook, the model has already been loaded!
+
+The guides do a great job [explaining](https://guides.emberjs.com/v2.17.0/routing/specifying-a-routes-model/#toc_dynamic-models) this feature.
+
+Before I started using JSON:API I thought this feature made a lot of sense, it would only fetch models when the application didn't have them. However, over the last year there's been a pattern I've seen in a couple Ember applications that this behavior complicates.
+
+Let's say we have a `posts.index` route that lists all of the posts. It's model hook:
+
+```js
+// app/routes/posts/index.js
+
+export default Ember.Route.extend({
+  model() {
+    return this.store.findAll('post');
+  }
+});
+```
+
+And then we also have a `posts.show` route that shows a specific post. When viewing a post we want to side load some additional data, like the post's comments.
+
+```js
+// app/routes/posts/show.js
+
+export default Ember.Route.extend({
+  model(params) {
+    return this.store.findRecord('post', params.post_id, {
+      include: ['comments.author'],
+      reload: true
+    });
+  }
+});
+```
+
+This creates an issue: Since the post model is already loaded, when we click a link on the index page to view a post, the model hook never runs. This causes the post to either render without comments, or worse, render with an async call to comments and then N+1 async calls to those comment's authors.
```

Comment:
It's not that much an issue with the record being loaded but being skipped due to passing the record and not it's ID to `<LinkTo>` it `transitionTo`. To be more precise the issue that another place in the application controls if the `model` hook run. This is very error prune. The model hook itself should control if it reruns.